rainbows.rb

```diff
@@ -1,29 +1,52 @@
-red = [255,0,0]
-green = [0,255,0]
-blue = [0,0,255]
-
-color_lookup = {
-  [255,0,0] => "red",
-  [255,125,0] => "orange",
-  [255,255,0] => "yellow",
-  [125,255,0] => "spring green",
-  [0,255,0] => "green",
-  [0,255,125] => "turquoise",
-  [0,255,255] => "cyan",
-  [0,125,255] => "ocean",
-  [0,0,255] => "blue",
-  [125,0,255] => "violet",
-  [255,0,255] => "magenta",
-  [255,0,125] => "raspberry"
-}
-
-def mix_colors(color1,color2)
-  mixed_color = []
-  3.times do |i|
-    mixed_color[i] = color1[i] + color2[i]
+require 'colorize'
+
+def get_color (red, green, blue)
+
+  rainbow = [
+    { red: 
+      { r: 255, g: 0, b: 0 } 
+    },
+
+    { orange: 
+      { r: 255, g: 127, b: 0 } 
+    },
+
+    { yellow: 
+      { r: 255, g: 255, b: 0 }
+    },
+
+    { green: 
+      { r: 0, g: 255, b: 0 }
+    },
+
+    { blue: 
+      { r: 0, g: 0, b: 255 }
+    },
+
+    { indigo: 
+      { r: 75, g: 0, b: 130 }
+    },
+    
+    { violet:
+     { r: 143, g: 0, b: 255 }
+    },
+  ]
+
+  rainbow.each do |color|
+    color.each do |key, value|
+      if red == value[:r] && green == value[:g] && blue == value[:b]
+        return key.to_s
+      else
+        puts "Sorry, I do not know that color!"
+      end
+    end
   end
-  return mixed_color
 end
 
-new_color = mix_colors(red,green)
-puts "#{color_lookup[red]} #{red} + #{color_lookup[green]} #{green} = #{color_lookup[new_color]} #{new_color}"
+puts get_color(255, 0, 0).colorize(:red)
+puts get_color(255, 127, 0).colorize(:orange)
+puts get_color(255, 255, 0).colorize(:yellow)
+puts get_color(0, 255, 0).colorize(:green)
+puts get_color(0, 0, 255).colorize(:blue)
+puts get_color(75, 0, 130).colorize(:indigo)
+puts get_color(143, 0, 255).colorize(:violet)
```

Comment:
This works, but I was hoping you all could figure out how to avoid manually setting the colors in colorize. Checkout how @DStorck did it!